### PR TITLE
Feature added : Support paginated topic list page

### PR DESCRIPTION
**Pull request by sajinidesilva**
- Improved topic list page performace by paginating getChildren method

### CODE REVIEW

**Comment by ramith** [comment]
@sajinidesilva  Please attach a jira for this.

File: components/andes/org.wso2.carbon.andes.event.core/src/main/java/org/wso2/carbon/andes/event/core/internal/subscription/registry/TopicManagerServiceImpl.java [hunk 1]
```diff
@@ -103,7 +110,33 @@ public TopicNode getTopicTree() throws EventBrokerException {
...
+            UserRegistry userRegistry =
+                    this.registryService.getGovernanceSystemRegistry(EventBrokerHolder.getInstance().getTenantId());
+            if (!userRegistry.resourceExists(topicPath)) {
+                userRegistry.put(topicPath, userRegistry.newCollection());
+            }
+            log.info("Before calling userregistry.get method");
+            Collection collection = (Collection) userRegistry.get(topicPath);
+            log.info("After calling userregistry.get method");
+            String[] paths = collection.getChildren(startIndex, numberOfTopicsPerRound);
+            log.info("Children Paths length: " + paths.length);
```
Comment (hemikak) [inline components/andes/org.wso2.carbon.andes.event.core/src/main/java/org/wso2/carbon/andes/event/core/internal/subscription/registry/TopicManagerServiceImpl.java]: Do we need these logs? :)

Comment (ramith) [inline components/andes/org.wso2.carbon.andes.event.core/src/main/java/org/wso2/carbon/andes/event/core/internal/subscription/registry/TopicManagerServiceImpl.java]: no we don't!

File: components/andes/org.wso2.carbon.andes.event.core/src/main/java/org/wso2/carbon/andes/event/core/internal/subscription/registry/TopicManagerServiceImpl.java [hunk 2]
```diff
@@ -113,40 +146,53 @@ public TopicNode getTopicTree() throws EventBrokerException {
...
-                            TopicNode childNode = new TopicNode(nodeName, childTopic);
-                            nodes.add(childNode);
-                            buildTopicTree(childNode, (Collection) childResource, userRegistry);
-                        }
-                    }
+        String[] children = paths;
+        if (children != null) {
+            List<TopicNode> nodes = new ArrayList<TopicNode>();
+            for (String childTopic : children) {
+                if (childTopic.endsWith("/")) {
```
Comment (ramith) [inline components/andes/org.wso2.carbon.andes.event.core/src/main/java/org/wso2/carbon/andes/event/core/internal/subscription/registry/TopicManagerServiceImpl.java]: as far as I remember there path separator used to be defined in reg constants .

File: components/andes/org.wso2.carbon.andes.event.ui/src/main/resources/web/topics/topic_list_ajaxprocessor.jsp [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+    ConfigurationContext configContext = (ConfigurationContext) config.getServletContext()
+            .getAttribute(CarbonConstants.CONFIGURATION_CONTEXT);
+
+    //Server URL which is defined in the server.xml
+    String serverURL = CarbonUIUtil.getServerURL(config.getServletContext(),
+            session) + "AndesEventAdminService.AndesEventAdminServiceHttpsSoap12Endpoint";
+    AndesEventAdminServiceStub stub = new AndesEventAdminServiceStub(configContext, serverURL);
+    TopicNode topicNode;
+    String html="";
```
Comment (ramith) [inline components/andes/org.wso2.carbon.andes.event.ui/src/main/resources/web/topics/topic_list_ajaxprocessor.jsp]: can we use a java.lang.StringBuilder?

File: components/andes/org.wso2.carbon.andes.event.ui/src/main/resources/web/topics/topics.jsp [hunk 4]
```diff
@@ -64,6 +64,8 @@
                     <td valign="top" style="width:200px;" class="leftBox">
                         <div class="treeControl" id="topicTree">
                             <%
+                                int TopicCountPerRound = 100;
```
Comment (ramith) [inline components/andes/org.wso2.carbon.andes.event.ui/src/main/resources/web/topics/topics.jsp]: wrong variable name! . topicCountPerRound?

File: components/andes/org.wso2.carbon.andes.event.ui/src/main/resources/web/topics/topics.jsp [hunk 5]
```diff
@@ -217,6 +250,157 @@
...
+              jQuery('#showMore').click(function() {
+                //send a ajax request here and set html equal to data recevide by ajax
+                jQuery.ajax({
+                           url:"topic_list_ajaxprocessor.jsp?startIndex="+startIndex + "&count=" + topicCountPerRound
+                            + "&topicPath=event/topics",
+                        data:{},
+                        type:"POST",
+                        success:function(data){
+                        data = data.trim();
+                        if(data == "No more topics to show"){
```
Comment (ramith) [inline components/andes/org.wso2.carbon.andes.event.ui/src/main/resources/web/topics/topics.jsp]: does checking for this kind of error message adheres to best practice ? to me it looks funky!

Comment (sajinidesilva) [inline components/andes/org.wso2.carbon.andes.event.ui/src/main/resources/web/topics/topics.jsp]: Yeah I agree. But with our current framework I don't find another way of doing this.

**Comment by sajinidesilva** [comment]
public jira : https://wso2.org/jira/browse/MB-1805

**Pull request merged**